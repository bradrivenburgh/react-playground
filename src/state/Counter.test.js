import React, { createElement } from 'react';
import ReactDOM from 'react-dom';
import Counter from './Counter';
import renderer from 'react-test-renderer';

describe('Counter component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Counter />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it.skip('renders the UI as expected', () => {
        const tree = renderer
            .create(<Counter count={123} step={3} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    //Not sure how to check for a change in state for count
    it.skip('increases the count', ()=> {
        const initialCount = Counter.this.handleButtonClick;
    });

});

