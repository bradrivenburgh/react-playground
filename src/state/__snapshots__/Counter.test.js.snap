// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders the UI as expected 1`] = `
<div>
  <p>
    The current count: 
    0
  </p>
  <button>
    Add 1
  </button>
</div>
`;
